recaptcha_enterprise/snippets/passwordLeakAssessment.js

```diff
@@ -0,0 +1,224 @@
+
+* If you want to extend this behavior to your own implementation/ languages,
+* make sure to perform the following steps:
+* 1. Hash the credentials (First 2 bytes of the result is the
+* 'lookupHashPrefix')
+* 2. Encrypt the hash (result = 'encryptedUserCredentialsHash')
+* 3. Get back the PasswordLeak information from
+* reCAPTCHA Enterprise Create Assessment.
+* 4. Decrypt the obtained 'credentials.getReencryptedUserCredentialsHash()'
+* with the same key you used for encryption.
+* 5. Check if the decrypted credentials are present in
+* 'credentials.getEncryptedLeakMatchPrefixesList()'.
+* 6. If there is a match, that indicates a credential breach.
+*/
+async function checkPasswordLeak(
+  projectId,
+  recaptchaSiteKey,
+  token,
+  action,
+  username,
+  password
+) {
+  // Instantiate the recaptcha-password-check-helpers library to perform the
+  // cryptographic functions.
+  // Create the request to obtain the hash prefix and encrypted credentials.
+  const verification = await PasswordCheckVerification.create(
+    username,
+    password
+  );
+
+  const lookupHashPrefix = Buffer.from(
+    verification.getLookupHashPrefix()
+  ).toString('base64');
+  const encryptedUserCredentialsHash = Buffer.from(
+    verification.getEncryptedUserCredentialsHash()
+  ).toString('base64');
+  console.log('Hashes created.');
+
+  // Pass the credentials to the createPasswordLeakAssessment() to get back
+  // the matching database entry for the hash prefix.
+  const credentials = await createPasswordLeakAssessment(
+    projectId,
+    recaptchaSiteKey,
+    token,
+    action,
+    lookupHashPrefix,
+    encryptedUserCredentialsHash
+  );
+
+  // Convert to appropriate input format.
+  const reencryptedUserCredentialsHash = Buffer.from(
+    credentials.reencryptedUserCredentialsHash.toString(),
+    'base64'
+  );
+
+  const encryptedLeakMatchPrefixes = credentials.encryptedLeakMatchPrefixes.map(
+    prefix => {
+      return Buffer.from(prefix.toString(), 'base64');
+    }
+  );
+
+  // Verify if the encrypted credentials are present in the obtained
+  // match list to check if the credential is leaked.
+  const isLeaked = verification
+    .verify(reencryptedUserCredentialsHash, encryptedLeakMatchPrefixes)
+    .areCredentialsLeaked();
+
+  console.log(`Is Credential leaked: ${isLeaked}`);
+}
+
+// Create a reCAPTCHA Enterprise assessment.
+// Returns: PrivatePasswordLeakVerification which contains
+// reencryptedUserCredentialsHash and credential breach database whose prefix
+// matches the lookupHashPrefix.
+async function createPasswordLeakAssessment(
+  projectId,
+  recaptchaSiteKey,
+  token,
+  action,
+  lookupHashPrefix,
+  encryptedUserCredentialsHash
+) {
+  // Build the assessment request.
+  const createAssessmentRequest = {
+    parent: `projects/${projectId}`,
+    assessment: {
+      // Set the properties of the event to be tracked.
+      event: {
+        siteKey: recaptchaSiteKey,
+        token: token,
```

Comment:
nit: the reCAPTCHA token is not mandatory for password check requests (in fact, it will count as an extra assessment call)

Reply:
Thanks for the info. I've removed it from the sample. 
Double-checking the docs, I see that the API request doesn't include the site key or the token. But I think it should be explicitly called out in the docs as well. Since, all the other recaptcha requests include both those params, it could be mistaken as a mandatory param for all requests involving recaptcha.
Plmk your thoughts. @serargz

Reply:
yeah, that's a good point. We should update the docs to make this clear.